pkg/registry/core/node/storage/storage_test.go

```diff
@@ -165,125 +187,58 @@ func TestShortNames(t *testing.T) {
-		node     api.Node
+		node     *api.Node
 		query    string
 		location string
 		err      error
-	}{
-		{
-			name: "proxyable hostname with default port",
-			node: api.Node{
-				ObjectMeta: metav1.ObjectMeta{Name: "node0"},
-				Status: api.NodeStatus{
-					Addresses: []api.NodeAddress{
-						{
-							Type:    api.NodeInternalIP,
-							Address: "10.0.0.1",
-						},
-					},
-				},
-			},
-			query:    "node0",
-			location: "10.0.0.1:10250",
-			err:      nil,
-		},
-		{
-			name: "proxyable hostname with kubelet port in query",
-			node: api.Node{
-				ObjectMeta: metav1.ObjectMeta{Name: "node0"},
-				Status: api.NodeStatus{
-					Addresses: []api.NodeAddress{
-						{
-							Type:    api.NodeInternalIP,
-							Address: "10.0.0.1",
-						},
-					},
-				},
-			},
-			query:    "node0:5000",
-			location: "10.0.0.1:5000",
-			err:      nil,
-		},
-		{
-			name: "proxyable hostname with kubelet port in status",
-			node: api.Node{
-				ObjectMeta: metav1.ObjectMeta{Name: "node0"},
-				Status: api.NodeStatus{
-					Addresses: []api.NodeAddress{
-						{
-							Type:    api.NodeInternalIP,
-							Address: "10.0.0.1",
-						},
-					},
-					DaemonEndpoints: api.NodeDaemonEndpoints{
-						KubeletEndpoint: api.DaemonEndpoint{
-							Port: 5000,
-						},
-					},
-				},
-			},
-			query:    "node0",
-			location: "10.0.0.1:5000",
-			err:      nil,
-		},
-		{
-			name: "non-proxyable hostname with default port",
-			node: api.Node{
-				ObjectMeta: metav1.ObjectMeta{Name: "node0"},
-				Status: api.NodeStatus{
-					Addresses: []api.NodeAddress{
-						{
-							Type:    api.NodeInternalIP,
-							Address: "127.0.0.1",
-						},
-					},
-				},
-			},
-			query:    "node0",
-			location: "",
-			err:      proxyutil.ErrAddressNotAllowed,
-		},
-		{
-			name: "non-proxyable hostname with kubelet port in query",
-			node: api.Node{
-				ObjectMeta: metav1.ObjectMeta{Name: "node0"},
-				Status: api.NodeStatus{
-					Addresses: []api.NodeAddress{
-						{
-							Type:    api.NodeInternalIP,
-							Address: "127.0.0.1",
-						},
-					},
-				},
-			},
-			query:    "node0:5000",
-			location: "",
-			err:      proxyutil.ErrAddressNotAllowed,
-		},
-		{
-			name: "non-proxyable hostname with kubelet port in status",
-			node: api.Node{
-				ObjectMeta: metav1.ObjectMeta{Name: "node0"},
-				Status: api.NodeStatus{
-					Addresses: []api.NodeAddress{
-						{
-							Type:    api.NodeInternalIP,
-							Address: "127.0.0.1",
-						},
-					},
-					DaemonEndpoints: api.NodeDaemonEndpoints{
-						KubeletEndpoint: api.DaemonEndpoint{
-							Port: 443,
-						},
-					},
-				},
-			},
-			query:    "node0",
-			location: "",
-			err:      proxyutil.ErrAddressNotAllowed,
-		},
+	}
+	var createTestCase = func(name string, node *api.Node, query string, location string, err error) testCase {
+		return testCase{
+			name,
+			node,
+			query,
+			location,
+			err,
+		}
+	}
+
+	testCases := []testCase{
+		createTestCase(
```

Comment:
I prefer to leave this as data, because fields can be named.  E.g. I think it is easier to read:

```
{{
  name: "proxyable hostname with default port",
  node: newNode("node0", setNodeIPAddress("10.0.0.1")),
  query: "node0",
  location: "10.0.0.1:10250"
}, {
  name: "proxyable hostname with kubelet port in query",
  node: newNode("node0", setNodeIPAddress("10.0.0.1")),
  query: "node0:5000",
  location: "10.0.0.1:5000",
}, {
  // ...
}}
```

Also note - I can leave out fields like `err` because it is nil in this case.

Reply:
Done.